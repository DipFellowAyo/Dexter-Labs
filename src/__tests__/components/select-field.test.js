import React from 'react'
import Select from '@material-ui/core/Select'
import FormControl from '@material-ui/core/FormControl'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'

import SelectField from '../../components/forms/select-field'

describe('Select Field Component', () => {
  let selectField

  beforeEach(() => {
    selectField = shallow(<SelectField />)
  })
  afterAll(() => jest.resetAllMocks())

  it('should render without crashing', () => {
    expect(selectField).toBeTruthy()
  })

  it('should match the snapshot', () => {
    expect(toJson(selectField)).toMatchSnapshot()
  })

  it('accepts `name` as props', () => {
    expect(selectField.props().name).toBeUndefined()
  })

  it('should render a `FormControl` component', () => {
    expect(selectField.find(FormControl)).toBeTruthy()
  })

  it('`FormControl` accepts a prop fullwidth', () => {
    expect(selectField.find(FormControl).props().fullWidth).toBeTruthy()
  })

  it('should render a `Select` component', () => {
    expect(selectField.find(Select)).toBeTruthy()
  })
})
