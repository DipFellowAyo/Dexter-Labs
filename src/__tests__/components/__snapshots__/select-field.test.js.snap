// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select Field Component should match the snapshot 1`] = `
<WithStyles(ForwardRef(FormControl))
  fullWidth={true}
>
  <WithStyles(ForwardRef(Select)) />
</WithStyles(ForwardRef(FormControl))>
`;
